src/components/ReportWelcomeText.tsx

```diff
@@ -1,114 +1,86 @@
                     <>
-                        <Text>{props.translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartOne')}</Text>
-                        <Text style={[styles.textStrong]}>{ReportUtils.getDisplayNameForParticipant(props.report.ownerAccountID)}</Text>
-                        <Text>{props.translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartTwo')}</Text>
-                        <Text style={[styles.textStrong]}>{ReportUtils.getPolicyName(props.report)}</Text>
-                        <Text>{props.translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartThree')}</Text>
+                        <Text>{translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartOne')}</Text>
+                        <Text style={[styles.textStrong]}>{ReportUtils.getDisplayNameForParticipant(report?.ownerAccountID)}</Text>
+                        <Text>{translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartTwo')}</Text>
+                        <Text style={[styles.textStrong]}>{ReportUtils.getPolicyName(report)}</Text>
+                        <Text>{translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartThree')}</Text>
                     </>
                 )}
                 {isChatRoom && (
                     <>
                         <Text>{roomWelcomeMessage.phrase1}</Text>
-                        {roomWelcomeMessage.showReportName && (
+                        {roomWelcomeMessage.showReportName && !!report?.reportID && (
                             <Text
                                 style={[styles.textStrong]}
-                                onPress={() => Navigation.navigate(ROUTES.REPORT_WITH_ID_DETAILS.getRoute(props.report.reportID))}
+                                onPress={() => Navigation.navigate(ROUTES.REPORT_WITH_ID_DETAILS.getRoute(report.reportID))}
                                 suppressHighlighting
                             >
-                                {ReportUtils.getReportName(props.report)}
+                                {ReportUtils.getReportName(report)}
                             </Text>
                         )}
                         {roomWelcomeMessage.phrase2 !== undefined && <Text>{roomWelcomeMessage.phrase2}</Text>}
                     </>
                 )}
                 {isDefault && (
                     <Text>
-                        <Text>{props.translate('reportActionsView.beginningOfChatHistory')}</Text>
-                        {_.map(displayNamesWithTooltips, ({displayName, pronouns, accountID}, index) => (
-                            <Text key={`${displayName}${pronouns}${index}`}>
+                        <Text>{translate('reportActionsView.beginningOfChatHistory')}</Text>
+                        {displayNamesWithTooltips.map(({displayName, pronouns, accountID}, index) => (
+                            <Text key={`${displayName}${pronouns}${accountID}`}>
```

Comment:
Not sure if you have already noticed this, but I changed
```jsx
<Text key={`${displayName}${pronouns}${index}`}>
```
to
```tsx
<Text key={`${displayName}${pronouns}${accountID}`}>
```
ESlint complained `Do not use Array index in keys` [react/no-array-index-key](https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/no-array-index-key.md)

I assume `accountID` will be unique inside this array, can you confirm it?

Reply:
@nikosamofa Another alternative (if `accountID` can't be used) is to suppress the rule with `// eslint-disable-next-line react/no-array-index-key`.

Reply:
Addressed by [73b8b7f](https://github.com/Expensify/App/pull/33251/commits/73b8b7f3345fda6344a09e58e6fa450eb9346ba0)